py/builtinimport.c

```diff
@@ -62,9 +65,15 @@ bool mp_obj_is_package(mp_obj_t module) {
 // will return whether the path is a file, directory, or doesn't exist.
 STATIC mp_import_stat_t stat_path_or_frozen(const char *path) {
     #if MICROPY_MODULE_FROZEN
-    mp_import_stat_t st = mp_frozen_stat(path);
-    if (st != MP_IMPORT_STAT_NO_EXIST) {
-        return st;
+    int frozen_type;
+    void *data;
+    // Only try and load as a frozen module if it starts with .frozen/.
+    const int frozen_path_prefix_len = strlen(MP_FROZEN_PATH_PREFIX);
+    if (strncmp(path, MP_FROZEN_PATH_PREFIX, frozen_path_prefix_len) == 0) {
+        mp_import_stat_t st = mp_find_frozen_module(path + frozen_path_prefix_len, &frozen_type, &data);
+        if (st != MP_IMPORT_STAT_NO_EXIST) {
+            return st;
```

Comment:
Jim and I did discuss this offline... allowing this to fall though and search the filesystem would impact performance of imports.  Any modules that are not frozen will need to hit the disk at least twice, once for the search for `.frozen/foo.py` (which will almost always fail) and then once for the search that succeeds (at least).  Disk access can be very slow so best to avoid this additional overhead.

If the user needs to intercept/emulate frozen code they can modify `sys.path`.  Eg add an entry `.frozen_emulated` which would hit the filesystem (and place that before or after `.frozen` depending on the desired behaviour).

Reply:
Sounds good to me, yes I did think it'd be inefficient and potentially confusing (user won't know if frozen or fs module used). 
There's the new docs entry about `.frozen` in `sys.path` which looks good, perhaps a note about `.frozen` folders being masked could be added to https://docs.micropython.org/en/latest/genrst/core_language.html#import

Reply:
That's a good point... @dpgeorge I think I meant to remove the `if (st != MP_IMPORT_STAT_NO_EXIST) {` -- i.e. if the path starts with `".frozen/"` then we should return the mp_find_frozen_module result unconditonally?

(And a similar change below)

Reply:
Yes indeed that if should be removed.